src/components/Generic/Dialogs/CreatePrinterDialog.vue

```diff
@@ -147,22 +192,46 @@ export default defineComponent({
     async isValid() {
       return await this.validationObserver.validate();
     },
+    async createPrinter(newPrinterData: CreatePrinter) {
+      await this.printersStore.createPrinter(newPrinterData);
+      this.$bus.emit(infoMessageEvent, `Printer ${newPrinterData.printerName} created`);
+    },
+    async updatePrinter(updatedPrinter: CreatePrinter) {
+      const printerId = updatedPrinter.id;
+
+      await this.printersStore.updatePrinter({
+        printerId: printerId as string,
+        updatedPrinter,
+      });
+
+      this.$bus.emit(infoMessageEvent, `Printer ${updatedPrinter.printerName} updated`);
+    },
     async submit() {
       if (!(await this.isValid())) return;
       const formData = this.formData();
       if (!formData) return;
-      const newPrinterData = PrintersService.convertCreateFormToPrinter(formData);
-      await this.printersStore.createPrinter(newPrinterData);
-      this.$bus.emit(infoMessageEvent, `Printer ${newPrinterData.printerName} created`);
-      this.printerCrudForm().resetForm();
-      this.validationObserver.reset();
+      const createPrinter = PrintersService.convertCreateFormToPrinter(formData);
+      if (this.isUpdating()) {
+        this.updatePrinter(createPrinter);
+      } else {
+        this.createPrinter(createPrinter);
+      }
       this.closeDialog();
     },
     closeDialog() {
       this.showChecksPanel = false;
       this.testPrinterStore.clearEvents();
       this.printerCrudForm().resetForm();
       this.dialogsStore.closeDialog(this.dialogId);
+      this.printersStore.updateDialogPrinter = undefined;
+      this.copyPasteConnectionString = "";
+    },
+
+    clearForm() {
```

Comment:
I feel like the close dialog and clear form methods are doing multiple specialistic things. We will clean this up soon!